agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java

```diff
@@ -29,26 +51,155 @@ public String encode(
+        List<MessageParameterTuple> tuples = new ArrayList<>();
+        AtomicInteger chunkNr = new AtomicInteger(1);
+        final String chunkContextId = ChunkContextIdService.generateChunkContextId();
+        messageChunks.forEach(
+            chunk -> {
+              final String messageIdForChunk = MessageIdService.generateMessageId();
+              final long sequenceNumberForChunk =
+                  SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse);
+
+              final MessageHeaderParameters header = new MessageHeaderParameters();
+              header.copy(messageHeaderParameters);
+              header.setApplicationMessageId(messageIdForChunk);
+              header.setApplicationMessageSeqNo(sequenceNumberForChunk);
+              Chunk.ChunkComponent.Builder chunkInfo = Chunk.ChunkComponent.newBuilder();
+              chunkInfo.setContextId(chunkContextId);
+              chunkInfo.setCurrent(chunkNr.get());
+              chunkInfo.setTotal(messageChunks.size());
+              chunkInfo.setTotalSize(wholeMessage.length());
+              header.setChunkInfo(chunkInfo.build());
+
+              final PayloadParameters payload = new PayloadParameters();
+              payload.copyFrom(payloadParameters);
+              payload.setValue(
+                  ByteString.copyFromUtf8(
+                      Base64.getEncoder().encodeToString(chunk.getBytes(StandardCharsets.UTF_8))));
+
+              tuples.add(new MessageParameterTuple(header, payload));
+
+              chunkNr.getAndIncrement();
+            });
+        return tuples;
+      } else {
+        getNativeLogger()
+            .debug(
+                "The message is not chunked since the current size of the payload ({}) is not above the limitation or the technical message type '{}' does not support chunking.",
+                payloadParameters.getValue().toStringUtf8().length(),
+                messageHeaderParameters.getTechnicalMessageType().getKey());
+        getNativeLogger()
+            .debug("The content is encoded, since in other cases the content is encoded as well.");
+        final PayloadParameters payload = new PayloadParameters();
+        payload.copyFrom(payloadParameters);
+        payload.setValue(
+            ByteString.copyFromUtf8(
+                Base64.getEncoder()
+                    .encodeToString(
+                        payloadParameters
+                            .getValue()
+                            .toStringUtf8()
+                            .getBytes(StandardCharsets.UTF_8))));
+        return Collections.singletonList(
+            new MessageParameterTuple(messageHeaderParameters, payload));
+      }
+    } else {
+      getNativeLogger()
+          .debug(
+              "The message type needs to be base64 encoded, therefore we are encoding the raw value.");
+      if (messageHeaderParameters.getTechnicalMessageType().getNeedsBase64Encoding()) {
```

Comment:
This line could cause an NPE as we are in the ELSE-Tree of 
` if (null != messageHeaderParameters.getTechnicalMessageType()
        && (...))`

Reply:
Right, this could be the case. But the ´getTechnicalMessageType´ is never ´null´ since it will cause an error within the validation of the parameters. I removed the additional ´null´ check.